src/main/java/org/isf/sms/service/SmsSender.java

```diff
@@ -57,77 +56,50 @@ public void run() {
 			List<Sms> smsList = null;
 			try {
 				smsList = smsOp.getList();
-			} catch (OHServiceException ohServiceException) {
-				LOGGER.error(ohServiceException.getMessage(), ohServiceException);
+			} catch (OHServiceException e1) {
+				e1.printStackTrace();
 			}
 			if (!smsList.isEmpty()) {
-				LOGGER.info("Found {} SMS to send", smsList.size());
-				if (SmsParameters.MODE.equals("GSM")) {
-					
-					SmsSenderGSM sender = new SmsSenderGSM();
-					if (sender.initialize()) {
-						for (Sms sms : smsList) {
-							if (sms.getSmsDateSched().before(new Date())) {
-								boolean result = sender.sendSMS(sms, GeneralData.DEBUG);
-								if (result) {
-									sms.setSmsDateSent(new Date());
-									try {
-										smsOp.saveOrUpdate(sms);
-									} catch (OHServiceException ohServiceException) {
-										LOGGER.error(ohServiceException.getMessage(), ohServiceException);
-									}
-									LOGGER.debug("Sent");
-								} else {
-									LOGGER.error("Not sent");
+				LOGGER.info("Found {} SMS to send", Integer.valueOf(smsList.size()));
+				SmsSenderOperations sender = Context.getApplicationContext().getBean(SmsSenderOperations.class);
+				if (sender.initialize()) {
+					for (Sms sms : smsList) {
+						if (sms.getSmsDateSched().before(new Date())) {
+							boolean result = sender.sendSMS(sms);
+							if (result) {
+								sms.setSmsDateSent(new Date());
+								try {
+									smsOp.saveOrUpdate(sms);
+								} catch (OHServiceException e) {
+									e.printStackTrace();
```

Comment:
the correct form is with `ohServiceException` and `LOGGER.error`

Reply:
corrected